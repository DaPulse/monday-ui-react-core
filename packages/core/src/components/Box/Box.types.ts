export type Disabled = "opacityDisabled";

export type Border = "border";

export type BorderColor = "borderColorUiBorderColor" | "borderColorLayoutBorderColor";

export type BackgroundColor =
  | "bgPrimaryBackgroundColor"
  | "bgSecondaryBackgroundColor"
  | "bgGreyBackgroundColor"
  | "bgAllgreyBackgroundColor"
  | "bgInvertedColorBackground";

export type BoxTextColor = "textPrimaryTextColor" | "textTextColorOnInverted" | "textSecondaryTextColor";

export type BoxPropSize = "xs" | "small" | "medium" | "large" | "xl" | "xxl" | "xxxl";

export type BoxPropSizeWithAuto = "auto" | BoxPropSize;

export type Shadow = Extract<BoxPropSize, "xs" | "small" | "medium" | "large">;

export type Rounded = "small" | "medium" | "big";
