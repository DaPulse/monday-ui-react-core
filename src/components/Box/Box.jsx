import React, { useRef, forwardRef } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import useMergeRefs from "hooks/useMergeRefs";
import styles from "./Box.module.scss";
import {
  DISABLED,
  BORDER,
  BORDER_TOP,
  BORDER_END,
  BORDER_BOTTOM,
  BORDER_START,
  BORDER_COLOR,
  ROUNDED,
  ROUNDED_TOP,
  ROUNDED_END,
  ROUNDED_BOTTOM,
  ROUNDED_START,
  SHADOW,
  MARGIN,
  MARGIN_X,
  MARGIN_Y,
  MARGIN_TOP,
  MARGIN_END,
  MARGIN_BOTTOM,
  MARGIN_START,
  PADDING,
  PADDING_X,
  PADDING_Y,
  PADDING_TOP,
  PADDING_END,
  PADDING_BOTTOM,
  PADDING_START,
  BACKGROUND_COLORS,
  COLORS
} from "./BoxConstants";

const Box = forwardRef(
  (
    {
      className,
      id,
      elementType,
      children,
      disabled,
      border,
      borderTop,
      borderEnd,
      borderBottom,
      borderStart,
      borderColor,
      rounded,
      roundedTop,
      roundedEnd,
      roundedBottom,
      roundedStart,
      shadow,
      margin,
      marginX,
      marginY,
      marginTop,
      marginEnd,
      marginBottom,
      marginStart,
      padding,
      paddingX,
      paddingY,
      paddingTop,
      paddingEnd,
      paddingBottom,
      paddingStart,
      textColor,
      backgroundColor
    },
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
    const Element = elementType;
    return (
      <Element
        ref={mergedRef}
        className={cx(
          styles.box,
          className,
          { [DISABLED.DISABLED]: disabled },
          border,
          borderTop,
          borderEnd,
          borderBottom,
          borderStart,
          borderColor,
          rounded,
          roundedTop,
          roundedEnd,
          roundedBottom,
          roundedStart,
          shadow,
          margin,
          marginX,
          marginY,
          marginTop,
          marginEnd,
          marginBottom,
          marginStart,
          padding,
          paddingX,
          paddingY,
          paddingTop,
          paddingEnd,
          paddingBottom,
          paddingStart,
          textColor,
          backgroundColor
        )}
        id={id}
      >
        {children}
      </Element>
    );
  }
);

Box.borders = BORDER;
Box.borderTops = BORDER_TOP;
Box.borderEnds = BORDER_END;
Box.borderBottoms = BORDER_BOTTOM;
Box.borderStarts = BORDER_START;
Box.borderColors = BORDER_COLOR;
Box.roundeds = ROUNDED;
Box.roundedTops = ROUNDED_TOP;
Box.roundedEnds = ROUNDED_END;
Box.roundedBottoms = ROUNDED_BOTTOM;
Box.roundedStarts = ROUNDED_START;
Box.shadows = SHADOW;
Box.margins = MARGIN;
Box.marginXs = MARGIN_X;
Box.marginYs = MARGIN_Y;
Box.marginTops = MARGIN_TOP;
Box.marginEnds = MARGIN_END;
Box.marginBottoms = MARGIN_BOTTOM;
Box.marginStarts = MARGIN_START;
Box.paddings = PADDING;
Box.paddingXs = PADDING_X;
Box.paddingYs = PADDING_Y;
Box.paddingTops = PADDING_TOP;
Box.paddingEnds = PADDING_END;
Box.paddingBottoms = PADDING_BOTTOM;
Box.paddingStarts = PADDING_START;
Box.backgroundColors = BACKGROUND_COLORS;
Box.textColors = COLORS;

Box.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  elementType: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ]),
  disabled: PropTypes.bool,
  border: PropTypes.oneOf([Box.borders.NONE, Box.borders.DEFAULT]),
  borderTop: PropTypes.oneOf([Box.borderTops.NONE, Box.borderTops.DEFAULT]),
  borderEnd: PropTypes.oneOf([Box.borderEnds.NONE, Box.borderEnds.DEFAULT]),
  borderBottom: PropTypes.oneOf([Box.borderBottoms.NONE, Box.borderBottoms.DEFAULT]),
  borderStart: PropTypes.oneOf([Box.borderStarts.NONE, Box.borderStarts.DEFAULT]),
  borderColor: PropTypes.oneOf([Box.borderColors.UI_BORDER_COLOR, Box.borderColors.LAYOUT_BORDER_COLOR]),
  rounded: PropTypes.oneOf([Box.roundeds.NONE, Box.roundeds.SMALL, Box.roundeds.MEDIUM, Box.roundeds.BIG]),
  roundedTop: PropTypes.oneOf([
    Box.roundedTops.NONE,
    Box.roundedTops.SMALL,
    Box.roundedTops.MEDIUM,
    Box.roundedTops.BIG
  ]),
  roundedEnd: PropTypes.oneOf([
    Box.roundedEnds.NONE,
    Box.roundedEnds.SMALL,
    Box.roundedEnds.MEDIUM,
    Box.roundedEnds.BIG
  ]),
  roundedBottom: PropTypes.oneOf([
    Box.roundedBottoms.NONE,
    Box.roundedBottoms.SMALL,
    Box.roundedBottoms.MEDIUM,
    Box.roundedBottoms.BIG
  ]),
  roundedStart: PropTypes.oneOf([
    Box.roundedStarts.NONE,
    Box.roundedStarts.SMALL,
    Box.roundedStarts.MEDIUM,
    Box.roundedStarts.BIG
  ]),
  shadow: PropTypes.oneOf([Box.shadows.XS, Box.shadows.SMALL, Box.shadows.MEDIUM, Box.shadows.LARGE]),
  margin: PropTypes.oneOf([
    Box.margins.NONE,
    Box.margins.AUTO,
    Box.margins.XS,
    Box.margins.SMALL,
    Box.margins.MEDIUM,
    Box.margins.LARGE,
    Box.margins.XL,
    Box.margins.XXL,
    Box.margins.XXXL
  ]),
  marginX: PropTypes.oneOf([
    Box.marginXs.NONE,
    Box.marginXs.AUTO,
    Box.marginXs.XS,
    Box.marginXs.SMALL,
    Box.marginXs.MEDIUM,
    Box.marginXs.LARGE,
    Box.marginXs.XL,
    Box.marginXs.XXL,
    Box.marginXs.XXXL
  ]),
  marginY: PropTypes.oneOf([
    Box.marginYs.NONE,
    Box.marginYs.AUTO,
    Box.marginYs.XS,
    Box.marginYs.SMALL,
    Box.marginYs.MEDIUM,
    Box.marginYs.LARGE,
    Box.marginYs.XL,
    Box.marginYs.XXL,
    Box.marginYs.XXXL
  ]),
  marginTop: PropTypes.oneOf([
    Box.marginTops.NONE,
    Box.marginTops.AUTO,
    Box.marginTops.XS,
    Box.marginTops.SMALL,
    Box.marginTops.MEDIUM,
    Box.marginTops.LARGE,
    Box.marginTops.XL,
    Box.marginTops.XXL,
    Box.marginTops.XXXL
  ]),
  marginEnd: PropTypes.oneOf([
    Box.marginEnds.NONE,
    Box.marginEnds.AUTO,
    Box.marginEnds.XS,
    Box.marginEnds.SMALL,
    Box.marginEnds.MEDIUM,
    Box.marginEnds.LARGE,
    Box.marginEnds.XL,
    Box.marginEnds.XXL,
    Box.marginEnds.XXXL
  ]),
  marginBottom: PropTypes.oneOf([
    Box.marginBottoms.NONE,
    Box.marginBottoms.AUTO,
    Box.marginBottoms.XS,
    Box.marginBottoms.SMALL,
    Box.marginBottoms.MEDIUM,
    Box.marginBottoms.LARGE,
    Box.marginBottoms.XL,
    Box.marginBottoms.XXL,
    Box.marginBottoms.XXXL
  ]),
  marginStart: PropTypes.oneOf([
    Box.marginStarts.NONE,
    Box.marginStarts.AUTO,
    Box.marginStarts.XS,
    Box.marginStarts.SMALL,
    Box.marginStarts.MEDIUM,
    Box.marginStarts.LARGE,
    Box.marginStarts.XL,
    Box.marginStarts.XXL,
    Box.marginStarts.XXXL
  ]),
  padding: PropTypes.oneOf([
    Box.paddings.NONE,
    Box.paddings.XS,
    Box.paddings.SMALL,
    Box.paddings.MEDIUM,
    Box.paddings.LARGE,
    Box.paddings.XL,
    Box.paddings.XXL,
    Box.paddings.XXXL
  ]),
  paddingX: PropTypes.oneOf([
    Box.paddingXs.NONE,
    Box.paddingXs.XS,
    Box.paddingXs.SMALL,
    Box.paddingXs.MEDIUM,
    Box.paddingXs.LARGE,
    Box.paddingXs.XL,
    Box.paddingXs.XXL,
    Box.paddingXs.XXXL
  ]),
  paddingY: PropTypes.oneOf([
    Box.paddingYs.NONE,
    Box.paddingYs.XS,
    Box.paddingYs.SMALL,
    Box.paddingYs.MEDIUM,
    Box.paddingYs.LARGE,
    Box.paddingYs.XL,
    Box.paddingYs.XXL,
    Box.paddingYs.XXXL
  ]),
  paddingTop: PropTypes.oneOf([
    Box.paddingTops.NONE,
    Box.paddingTops.XS,
    Box.paddingTops.SMALL,
    Box.paddingTops.MEDIUM,
    Box.paddingTops.LARGE,
    Box.paddingTops.XL,
    Box.paddingTops.XXL,
    Box.paddingTops.XXXL
  ]),
  paddingEnd: PropTypes.oneOf([
    Box.paddingEnds.NONE,
    Box.paddingEnds.XS,
    Box.paddingEnds.SMALL,
    Box.paddingEnds.MEDIUM,
    Box.paddingEnds.LARGE,
    Box.paddingEnds.XL,
    Box.paddingEnds.XXL,
    Box.paddingEnds.XXXL
  ]),
  paddingBottom: PropTypes.oneOf([
    Box.paddingBottoms.NONE,
    Box.paddingBottoms.XS,
    Box.paddingBottoms.SMALL,
    Box.paddingBottoms.MEDIUM,
    Box.paddingBottoms.LARGE,
    Box.paddingBottoms.XL,
    Box.paddingBottoms.XXL,
    Box.paddingBottoms.XXXL
  ]),
  paddingStart: PropTypes.oneOf([
    Box.paddingStarts.NONE,
    Box.paddingStarts.XS,
    Box.paddingStarts.SMALL,
    Box.paddingStarts.MEDIUM,
    Box.paddingStarts.LARGE,
    Box.paddingStarts.XL,
    Box.paddingStarts.XXL,
    Box.paddingStarts.XXXL
  ]),
  textColor: PropTypes.oneOf([
    Box.textColors.PRIMARY_COLOR,
    Box.textColors.PRIMARY_HOVER_COLOR,
    Box.textColors.PRIMARY_SELECTED_COLOR,
    Box.textColors.PRIMARY_TEXT_COLOR,
    Box.textColors.TEXT_COLOR_ON_PRIMARY,
    Box.textColors.TEXT_COLOR_ON_INVERTED,
    Box.textColors.SECONDARY_TEXT_COLOR,
    Box.textColors.PLACEHOLDER_COLOR,
    Box.textColors.ICON_COLOR,
    Box.textColors.LINK_COLOR,
    Box.textColors.DISABLED_TEXT_COLOR
  ]),
  backgroundColor: PropTypes.oneOf([
    Box.backgroundColors.PRIMARY_BACKGROUND_COLOR,
    Box.backgroundColors.PRIMARY_HOVER_BACKGROUND_COLOR,
    Box.backgroundColors.SECONDARY_BACKGROUND_COLOR,
    Box.backgroundColors.GREY_BACKGROUND_COLOR,
    Box.backgroundColors.ALL_GREY_BACKGROUND_COLOR,
    Box.backgroundColors.INVERTED_COLOR_BACKGROUND,
    Box.backgroundColors.DISABLED_BACKGROUND_COLOR
  ])
};
Box.defaultProps = {
  className: "",
  id: undefined,
  elementType: "div",
  children: undefined,
  disabled: undefined,
  border: undefined,
  borderTop: undefined,
  borderEnd: undefined,
  borderBottom: undefined,
  borderStart: undefined,
  borderColor: undefined,
  rounded: undefined,
  roundedTop: undefined,
  roundedEnd: undefined,
  roundedBottom: undefined,
  roundedStart: undefined,
  shadow: undefined,
  margin: undefined,
  marginX: undefined,
  marginY: undefined,
  marginTop: undefined,
  marginEnd: undefined,
  marginBottom: undefined,
  marginStart: undefined,
  padding: undefined,
  paddingX: undefined,
  paddingY: undefined,
  paddingTop: undefined,
  paddingEnd: undefined,
  paddingBottom: undefined,
  paddingStart: undefined,
  textColor: undefined,
  backgroundColor: undefined
};

export default Box;
