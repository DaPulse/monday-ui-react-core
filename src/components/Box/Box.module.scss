@import "../../styles/themes";

// Monday ui styles dependencies
@import "~monday-ui-style/src/common";

// Utilities map
@import "../../styles/utilities/utilities-settings";


// Generating Utility class locally for Box component
@include utility-class-factory($utilities-modules, $export: "modules");

