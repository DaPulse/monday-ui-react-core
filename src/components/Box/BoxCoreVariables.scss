// BOX Core: variables
// Mock Custom properties that should be imported from "monday-ui-styles"

:root {
  --border-width: 1px;
  --border-style: solid;
}

:root,
.light-app-theme,
.default-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px rgba(0, 0, 0, 0.1);
}

.dark-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px rgba(9, 11, 25, 0.5);
}

.black-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px #000000;
}

.hacker-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px #000000;
}