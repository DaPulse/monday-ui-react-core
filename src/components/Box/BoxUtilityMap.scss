@use "sass:map";

$utilities-modules: (
        "opacity": (
                property: opacity,
                values: (
                        disabled: var(--disabled-component-opacity),
                )
        ),
        "border": (
                property: border,
                values: (
                        null: var(--border-width) var(--border-style) var(--ui-border-color),
                        0: 0,
                )
        ),
        "borderTop": (
                property: border-top,
                class: borderTop,
                values: (
                        null: var(--border-width) var(--border-style) var(--ui-border-color),
                        0: 0,
                )
        ),
        "borderEnd": (
                property: border-inline-end,
                class: borderEnd,
                values: (
                        null: var(--border-width) var(--border-style) var(--ui-border-color),
                        0: 0,
                )
        ),
        "borderBottom": (
                property: border-bottom,
                class: borderBottom,
                values: (
                        null: var(--border-width) var(--border-style) var(--ui-border-color),
                        0: 0,
                )
        ),
        "borderStart": (
                property: border-inline-start,
                class: borderStart,
                values: (
                        null: var(--border-width) var(--border-style) var(--ui-border-color),
                        0: 0,
                )
        ),
        "borderColor": (
                property: border-color,
                class: borderColor,
                values: $core-border-colors-vars,
        ),
        "rounded": (
                property: border-radius,
                class: rounded,
                values: (
                        0: 0,
                        small: var(--border-radius-small),
                        medium: var(--border-radius-medium),
                        big: var(--border-radius-big),
                ),
        ),
        "roundedTop": (
                property: border-top-left-radius border-top-right-radius,
                class: roundedTop,
                values: (
                        0: 0,
                        small: var(--border-radius-small),
                        medium: var(--border-radius-medium),
                        big: var(--border-radius-big),
                ),
        ),
        "roundedEnd": (
                property: border-top-right-radius border-bottom-right-radius,
                class: roundedEnd,
                values: (
                        0: 0,
                        small: var(--border-radius-small),
                        medium: var(--border-radius-medium),
                        big: var(--border-radius-big),
                ),
        ),
        "roundedBottom": (
                property: border-bottom-right-radius border-bottom-left-radius,
                class: roundedBottom,
                values: (
                        0: 0,
                        small: var(--border-radius-small),
                        medium: var(--border-radius-medium),
                        big: var(--border-radius-big),
                ),
        ),
        "roundedStart": (
                property: border-bottom-left-radius border-top-left-radius,
                class: roundedStart,
                values: (
                        0: 0,
                        small: var(--border-radius-small),
                        medium: var(--border-radius-medium),
                        big: var(--border-radius-big),
                ),
        ),
        "shadow": (
                property: box-shadow,
                class: shadow,
                values: (
                        xs: var(--box-shadow-xs),
                        small: var(--box-shadow-small),
                        medium: var(--box-shadow-medium),
                        large: var(--box-shadow-large),
                )
        ),
        "margin": (
                property: margin,
                class: m,
                values: map.merge($core-spacing-vars, (auto: auto)),
        ),
        "marginX": (
                property: margin-right margin-left,
                class: mx,
                values: map.merge($core-spacing-vars, (auto: auto))
        ),
        "marginY": (
                property: margin-top margin-bottom,
                class: my,
                values: map.merge($core-spacing-vars, (auto: auto)),
        ),
        "marginTop": (
                property: margin-top,
                class: mt,
                values: map.merge($core-spacing-vars, (auto: auto)),
        ),
        "marginEnd": (
                property: margin-inline-end,
                class: me,
                values: map.merge($core-spacing-vars, (auto: auto)),
        ),
        "marginBottom": (
                property: margin-bottom,
                class: mb,
                values: map.merge($core-spacing-vars, (auto: auto)),
        ),
        "marginStart": (
                property: margin-inline-start,
                class: ms,
                values: map.merge($core-spacing-vars, (auto: auto)),
        ),
        "padding": (
                property: padding,
                class: p,
                values: $core-spacing-vars,
        ),
        "paddingX": (
                property: padding-right padding-left,
                class: px,
                values: $core-spacing-vars,
        ),
        "paddingY": (
                property: padding-top padding-bottom,
                class: py,
                values: $core-spacing-vars,
        ),
        "paddingTop": (
                property: padding-top,
                class: pt,
                values: $core-spacing-vars
        ),
        "paddingEnd": (
                property: padding-inline-end,
                class: pe,
                values: $core-spacing-vars
        ),
        "paddingBottom": (
                property: padding-bottom,
                class: pb,
                values: $core-spacing-vars
        ),
        "paddingStart": (
                property: padding-inline-start,
                class: ps,
                values: $core-spacing-vars
        ),
        "color": (
                property: color,
                class: text,
                values: $core-colors-vars,
        ),
        "backgroundColor": (
                property: background-color,
                class: bg,
                values: map.merge(
                                $core-background-colors-vars,
                                (
                                        "transparent": transparent
                                ),
                ),
        ),
);
