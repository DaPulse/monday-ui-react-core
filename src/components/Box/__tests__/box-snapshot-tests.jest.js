import React from "react";
import renderer from "react-test-renderer";
import Box from "../Box";

/**
 * There are cases where the component we want to test in the snapshot test will contain additional components.
 We do not want changes to the additional components to fail our component snapshot's test.
 Therefore, we will replace the instances of the other external components in the snapshot test with mock/stub components in these cases.
 */

/** example for external library
 jest.mock("react-transition-group", () => {
  const FakeTransition = jest.fn(({ children }) => children);
  const FakeSwitchTransition = jest.fn(({ children }) => children);
  const FakeCSSTransition = jest.fn(({ children }) => children);

  // We return here the instance of the mock / stub library object content
  return {
    CSSTransition: FakeCSSTransition,
    Transition: FakeTransition,
    SwitchTransition: FakeSwitchTransition
  };
});
 **/

/** example for internal component
jest.mock("../../Button/Button", () => {
  // We return here the instance of the mock / stub component
  return ({ onClick }) => (
    <div data-testid="cancel-button" {...(onClick && { "data-onclick": "onclick-provided" })} />
  );
});
**/

describe("Box renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<Box></Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with border prop", () => {
    const tree = renderer.create(<Box border={Box.borders.DEFAULT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderTop prop", () => {
    const tree = renderer.create(<Box borderTop={Box.borderTops.DEFAULT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderEnd prop", () => {
    const tree = renderer.create(<Box borderEnd={Box.borderEnds.DEFAULT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderBottom prop", () => {
    const tree = renderer.create(<Box borderBottom={Box.borderBottoms.DEFAULT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderStart prop", () => {
    const tree = renderer.create(<Box borderStart={Box.borderStarts.DEFAULT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderColor prop", () => {
    const tree = renderer.create(<Box borderColor={Box.borderColors.UI_BORDER_COLOR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with rounded prop", () => {
    const tree = renderer.create(<Box rounded={Box.roundeds.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with roundedTop prop", () => {
    const tree = renderer.create(<Box roundedTop={Box.roundedTops.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with roundedEnd prop", () => {
    const tree = renderer.create(<Box roundedTop={Box.roundedTops.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with roundedBottom prop", () => {
    const tree = renderer.create(<Box roundedTop={Box.roundedTops.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with roundedStart prop", () => {
    const tree = renderer.create(<Box roundedStart={Box.roundedStarts.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with shadow prop", () => {
    const tree = renderer.create(<Box shadow={Box.shadows.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with margin prop", () => {
    const tree = renderer.create(<Box margin={Box.margins.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginX prop", () => {
    const tree = renderer.create(<Box marginX={Box.marginXs.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginY prop", () => {
    const tree = renderer.create(<Box marginY={Box.marginYs.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginTop prop", () => {
    const tree = renderer.create(<Box marginTop={Box.marginTops.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginEnd prop", () => {
    const tree = renderer.create(<Box marginEnd={Box.marginEnds.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginStart prop", () => {
    const tree = renderer.create(<Box marginStart={Box.marginStarts.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with padding prop", () => {
    const tree = renderer.create(<Box padding={Box.paddings.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingX prop", () => {
    const tree = renderer.create(<Box paddingX={Box.paddingXs.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingY prop", () => {
    const tree = renderer.create(<Box paddingY={Box.paddingYs.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingTop prop", () => {
    const tree = renderer.create(<Box paddingTop={Box.paddingTops.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingEnd prop", () => {
    const tree = renderer.create(<Box paddingEnd={Box.paddingEnds.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingBottom prop", () => {
    const tree = renderer.create(<Box paddingBottom={Box.paddingBottoms.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingStart prop", () => {
    const tree = renderer.create(<Box paddingStart={Box.paddingStarts.SMALL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with backgronudColor prop", () => {
    const tree = renderer.create(<Box BackgronudColor={Box.backgroundColors.PRIMARY_BACKGROUND_COLOR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with text prop", () => {
    const tree = renderer.create(<Box text={Box.texts.PRIMARY_COLOR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
