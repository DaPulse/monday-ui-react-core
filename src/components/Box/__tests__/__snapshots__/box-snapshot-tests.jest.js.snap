// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly with backgronudColor prop 1`] = `
<div
  className="box"
/>
`;

exports[`Box renders correctly with border prop 1`] = `
<div
  className="box border"
/>
`;

exports[`Box renders correctly with borderBottom prop 1`] = `
<div
  className="box borderBottom"
/>
`;

exports[`Box renders correctly with borderColor prop 1`] = `
<div
  className="box borderColorUiBorderColor"
/>
`;

exports[`Box renders correctly with borderEnd prop 1`] = `
<div
  className="box borderEnd"
/>
`;

exports[`Box renders correctly with borderStart prop 1`] = `
<div
  className="box borderStart"
/>
`;

exports[`Box renders correctly with borderTop prop 1`] = `
<div
  className="box borderTop"
/>
`;

exports[`Box renders correctly with empty props 1`] = `
<div
  className="box"
/>
`;

exports[`Box renders correctly with margin prop 1`] = `
<div
  className="box mSmall"
/>
`;

exports[`Box renders correctly with marginEnd prop 1`] = `
<div
  className="box meSmall"
/>
`;

exports[`Box renders correctly with marginStart prop 1`] = `
<div
  className="box msSmall"
/>
`;

exports[`Box renders correctly with marginTop prop 1`] = `
<div
  className="box mtSmall"
/>
`;

exports[`Box renders correctly with marginX prop 1`] = `
<div
  className="box mxSmall"
/>
`;

exports[`Box renders correctly with marginY prop 1`] = `
<div
  className="box mySmall"
/>
`;

exports[`Box renders correctly with padding prop 1`] = `
<div
  className="box pSmall"
/>
`;

exports[`Box renders correctly with paddingBottom prop 1`] = `
<div
  className="box pbSmall"
/>
`;

exports[`Box renders correctly with paddingEnd prop 1`] = `
<div
  className="box peSmall"
/>
`;

exports[`Box renders correctly with paddingStart prop 1`] = `
<div
  className="box psSmall"
/>
`;

exports[`Box renders correctly with paddingTop prop 1`] = `
<div
  className="box ptSmall"
/>
`;

exports[`Box renders correctly with paddingX prop 1`] = `
<div
  className="box pxSmall"
/>
`;

exports[`Box renders correctly with paddingY prop 1`] = `
<div
  className="box pySmall"
/>
`;

exports[`Box renders correctly with rounded prop 1`] = `
<div
  className="box roundedSmall"
/>
`;

exports[`Box renders correctly with roundedBottom prop 1`] = `
<div
  className="box roundedTopSmall"
/>
`;

exports[`Box renders correctly with roundedEnd prop 1`] = `
<div
  className="box roundedTopSmall"
/>
`;

exports[`Box renders correctly with roundedStart prop 1`] = `
<div
  className="box roundedStartSmall"
/>
`;

exports[`Box renders correctly with roundedTop prop 1`] = `
<div
  className="box roundedTopSmall"
/>
`;

exports[`Box renders correctly with shadow prop 1`] = `
<div
  className="box shadowSmall"
/>
`;

exports[`Box renders correctly with text prop 1`] = `
<div
  className="box textPrimaryColor"
/>
`;
